news-explorer-frontend/src/components/App/App.jsx

```diff
@@ -0,0 +1,342 @@
+            console.log("card-id", data._id);
+          })
+          //.then(console.log("savedArticle =>" , savedArticle))
+          .catch((err) => console.log("bookmark Error =>", err));
+    }
+
+    // useEffect(() =>  {
+    //   console.log("savedArticle =>" , savedArticle)
+    // }, [savedArticle])
+
+     //remove bookmark
+    function deleteCardFromSaved(card) {
+      api
+        .deleteArticle(card.id)
+          .then((res) => {
+            console.log("deleted card props.id ...>", res._id )
+            const newSavedArticles = savedArticle.filter((card) => card._id !== res._id);
+          setSavedArticle(newSavedArticles);
+          console.log("savedArticles==>", savedArticle);
+
+        })
+        .catch((err) => console.log("delete function error =>", err))
+    }
+
+    function searchItems(searchValue) {
+      if(searchValue !== '') {  // console.log(searchValue);
+        setIsLoading(true);
+    
+          newsApi.getNews(searchValue)
+            .then( (res) => {   
+              console.log("res--->", res)
+              let searchResult = res.articles.filter((item) => {
+                return Object.values(item)
+                  .join('')
+                  .toLowerCase()
+                  .includes(searchValue.toLowerCase());
+              })
+              setFilteredResults(searchResult)
+            })
+            .finally(() => {
+              setIsLoading(false);
+            })          
+      } else { 
+        console.log("nothing found....");
+        setFilteredResults('')
+      }
+    }
+
+    useEffect(() => {
+      searchItems(keyword);
+    }, [keyword]);
+
+
+  useEffect(() => {
+    const closeByEscape = (e) => {
+      if (e.key === 'Escape') {
+        closeAllPopups();
+      }
+    };
+    const closeByModal = (e) => {
+      if (e.target.classList.contains("popup_open")) {
+        closeAllPopups();
+      }
+    };
+
+    document.addEventListener('keydown', closeByEscape);
+    document.addEventListener('mousedown', closeByModal);
+
+    return () => {
+      document.removeEventListener('keydown', closeByEscape);
+      document.removeEventListener('click', closeByModal);
+    }
+  }, []);
+
+  function handleSignInClick() {
+    closeAllPopups();
+    setIsSignInPopupOpen(true);
+  }
+
+  function handleSignUpClick() {
+    setIsSignUpPopupOpen(true);
+  }
+
+  function handleRegisterSuccess() {
+    closeAllPopups();
+    setIsRegisterPopupOpen(true);
+  }
+
+  function closeAllPopups() {
+    setIsSignInPopupOpen(false);
+    setIsSignUpPopupOpen(false);
+    setIsRegisterPopupOpen(false);
+    setIsMobileClicked(false);
+    setIsMobileMenuClicked(false);
+  }
+
+  function handleMobileClick () {
+   !isLoggedIn && setIsMobileClicked(true);
+  }
+
+  function handleMobileMenuClick() {
+    isLoggedIn && setIsMobileMenuClicked(true);
+  }
+
+  function handleSubmitClicked(){
+    setIsLoading(true);
+  } 
+
+  let location = useLocation();

```

Comment:
If a variable is declared using `let`, its value must change somewhere.
If a variable's value doesn’t change, it is declared using `const`.

**All your variables** should be declared using `const`

Reply:
fixed+